### PR TITLE
Enforce id requirement in TextField component

### PR DESCRIPTION
**Pull request by javiwq**
Closes issue #590 , follows up PR #681

### CODE REVIEW

File: app/frontend/src/components/TextField.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {
   TextField as MuiTextField,
   TextFieldProps,
 } from "@material-ui/core";
+import { BaseTextFieldProps } from "@material-ui/core/TextField"
```
Comment (darrenvong) [inline app/frontend/src/components/TextField.tsx]:
```suggestion
import { BaseTextFieldProps } from "@material-ui/core/TextField";
```

File: app/frontend/src/components/TextField.tsx [hunk 2]
```diff
@@ -23,12 +24,17 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+ interface AccessibleTextFieldProps extends BaseTextFieldProps {
+   id: BaseTextFieldProps["id"];
+   onChange?: TextFieldProps["onChange"]
```
Comment (darrenvong) [inline app/frontend/src/components/TextField.tsx]:
```suggestion
   onChange?: TextFieldProps["onChange"];
```

**Review by darrenvong — CHANGES_REQUESTED** [review]
Looks good apart from the nitpick on formatting I saw. It's worth running `yarn format` in future to fix these first before pushing anything :)

**Comment by javiwq** [comment]
Will do, thank you!

**Review by darrenvong — APPROVED** [review]

**Pull request merged**